[Network Packet Capture] Convert Dashboards to lens, add pipeline errors and format_versions

P1llus opened the pull request:
## What does this PR do?

This is part of a bigger cleanup that relates to https://github.com/elastic/security-team/issues/5910.

The tasks, when applicable was:

- [x] Update format_version to 2.6.0+
- [x] Update on_failure to include proper event.kind.
- [x] Update on_failure error message to be more descriptive
- [x] Add processor tags for on_failure message
- [x] If system/pipeline tests are missing, add them when possible. (Removed one broken)
- [x] If dashboards exist, replace with lens when needed, update dashboards if applicable.

Dashboard changes:
1. All table of contents.
2. Convert all to lens.
3. Add/remove/rename certain visualizations to fit the layout and usefulness better
4. Changed layout a bit.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Testing
**Use commit https://github.com/elastic/integrations/pull/6117/commits/70d11d1cbffccee75c2bf7d6d7552bb67801e64e for testing**
I have temporarily added interface options to each datastream to help with testing, it will be removed before merging. The steps to test this is:

1. Copy the PCAP files not including the folder, from the `_dev/deploy/docker/pcaps/*` folder to `~/.elastic-package/tmp/service_logs/*`
2. Bring up the stack (version 8.6.2).
3. Once everything is up and healthy, you can run the below snippet in Kibana Dev tools. It will update the agent policy to enable all protocols, and point them to their respective pcap files.
One note, is that since we have to use a direct reference to the pcap files, if you want to ingest more test-data, you would have to manually update the agent policy, and change the path to another pcap, save, wait for the ingest to finish, then change it again etc.

EDIT: Updated step 1, to make it more specific.

Test policy:
```
POST kbn:/api/fleet/package_policies
{
  "policy_id": "elastic-agent-managed-ep",
  "package": {
    "name": "network_traffic",
    "version": "1.16.0"
  },
  "name": "network_traffic-1",
  "description": "",
  "namespace": "default",
  "inputs": {
    "network-packet": {
      "enabled": true,
      "vars": {
        "never_install": false
      },
      "streams": {
        "network_traffic.amqp": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/amqp_channel_error.pcap",
            "port": [
              5672
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.cassandra": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/cassandra_mixed_frame.pcap",
            "port": [
              9042
            ],
            "geoip_enrich": true,
            "ignored_ops": [],
            "tags": []
          }
        },
        "network_traffic.dhcpv4": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/dhcp.pcap",
            "port": [
              67,
              68
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.dns": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/dns_ns.pcap",
            "port": [
              53
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.flow": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/http_10_connection_close.pcap",
            "geoip_enrich": true,
            "period": "10s",
            "timeout": "30s",
            "tags": []
          }
        },
        "network_traffic.http": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/http_10_connection_close.pcap",
            "port": [
              80,
              8080,
              8000,
              5000,
              8002
            ],
            "hide_keywords": [],
            "send_headers": [],
            "redact_headers": [],
            "include_body_for": [],
            "include_request_body_for": [],
            "include_response_body_for": [],
            "tags": []
          }
        },
        "network_traffic.icmp": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/icmp4_ping_over_vlan.pcap",
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.memcached": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/memcache_bin_tcp_stats.pcap",
            "port": [
              11211
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.mongodb": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/mongo_3.0_session.pcap",
            "port": [
              27017
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.mysql": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/mysql_long.pcap",
            "port": [
              3306,
              3307
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.nfs": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/nfs_v4.pcap",
            "port": [
              2049
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.pgsql": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/pgsql_long_result.pcap",
            "port": [
              5432
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.redis": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/redis_session.pcap",
            "port": [
              "6380"
            ],
            "geoip_enrich": true,
            "tags": []
          }
        },
        "network_traffic.sip": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/sip.pcap",
            "port": [
              5060
            ],
            "geoip_enrich": true,
            "use_tcp": false,
            "tags": []
          }
        },
        "network_traffic.thrift": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/thrift_integration.pcap",
            "port": [
              9090
            ],
            "geoip_enrich": true,
            "idl_files": [],
            "tags": []
          }
        },
        "network_traffic.tls": {
          "enabled": true,
          "vars": {
            "interface": "/tmp/service_logs/tls-version-13.pcap",
            "port": [
              443,
              993,
              995,
              5223,
              8443,
              8883,
              9243
            ],
            "geoip_enrich": true,
            "fingerprints": [],
            "tags": []
          }
        }
      }
    }
  }
}
```


## Related issues

- Relates https://github.com/elastic/integrations/pull/6106

## Screenshots (Somewhat limited due to test data available, but working fine).

Overview
![newOverview1](https://github.com/elastic/integrations/assets/8027539/7a60c4bd-ffee-45b1-a8da-d5a60fbc3c7e)
![newOverview2](https://github.com/elastic/integrations/assets/8027539/30e74b6a-9bba-4e1c-a32d-9583d6c67ce5)

Mysql:
![newMYSQL1](https://github.com/elastic/integrations/assets/8027539/94f2fc6b-49d2-4bb7-8aed-0dc9c610db37)
![newMYSQL2](https://github.com/elastic/integrations/assets/8027539/7fd249fe-d933-4937-acf6-0cbdb2350ad5)

NFS:
![newNFS1](https://github.com/elastic/integrations/assets/8027539/d052f6f7-8dda-4e52-adab-8e5623b4f50e)
![newNFS2](https://github.com/elastic/integrations/assets/8027539/87810de5-02a2-4f78-97b6-13a3ba2ffcc1)

PGSQL:
![newpg1](https://github.com/elastic/integrations/assets/8027539/6fd70c2b-e250-4aa9-89d1-e6453c8a1112)
![newpg2](https://github.com/elastic/integrations/assets/8027539/907c4f2a-582f-47fe-aa59-8cb45b44f974)

Thrift:
![newThrift1](https://github.com/elastic/integrations/assets/8027539/40dabe75-b010-4503-97e4-8804589496d6)
![newThrift2](https://github.com/elastic/integrations/assets/8027539/0c7ecd53-aa25-4182-9334-48bf4c4b0013)

Cassandra:
![newCassandra1](https://github.com/elastic/integrations/assets/8027539/4f5dd456-247f-40e2-a0f2-e1210b2025fe)

DHCP:
![newDHCP1](https://github.com/elastic/integrations/assets/8027539/13d6a180-27c2-4672-ba35-e2cac67393b6)

DNS:
![newDNS1](https://github.com/elastic/integrations/assets/8027539/11dd9581-56b9-4c62-ad9c-0fbe4714411c)
![newDNS2](https://github.com/elastic/integrations/assets/8027539/afc28652-18cc-4c06-b266-7377d0f123e3)

HTTP:
![newHTTP1](https://github.com/elastic/integrations/assets/8027539/c95c8b75-246a-498e-9ee1-2ec21703a345)
![newHTTP2](https://github.com/elastic/integrations/assets/8027539/476bc8e0-289c-4e50-82ec-0c06f49b7326)

MongoDB:
![newMongo1](https://github.com/elastic/integrations/assets/8027539/b8c7bba6-289e-415b-a522-fcfe2c6898aa)
![newMongo2](https://github.com/elastic/integrations/assets/8027539/9884405e-d91c-4ed6-9d23-aa7bf65370b2)
![newMongo3](https://github.com/elastic/integrations/assets/8027539/1c93298e-b620-49a7-b498-c61161d837f7)

P1llus commented:
Initial Draft added everything except dashboards, next commits will add them in, will keep it in draft for now

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6117/detail/PR-6117/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6117/detail/PR-6117/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6117/detail/PR-6117/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6117/detail/PR-6117/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_6117.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2023-05-08T10:32:33.848Z&rangeTo=2023-05-08T10:52:33.848Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=bc04afe18366d3f7d8dfd580796d5352&transactionId=f59db87c6500fffe)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-05-08T10:42:33.848+0000


* Duration: 58 min 4 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 177  |
| Skipped      | 0 |
| Total        | 177   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`0/0`) | :green_heart: 
Files | 100.0% (`0/0`) | :green_heart: 
Classes | 100.0% (`0/0`) | :green_heart: 
Methods | 75.0% (`48/64`) | :-1: -25.0
Lines | 100.0% (`0/0`) | :green_heart: 7.128
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

elasticmachine commented:
Pinging @elastic/security-external-integrations (Team:Security-External Integrations)

P1llus commented:
This does not include data views yet, to fix field collisions with Cloud Securiry Posture, will look into that before merging.

P1llus commented on packages/network_traffic/data_stream/amqp/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/dns/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/flow/manifest.yml:
```diff
@@ -6,6 +6,11 @@ streams:
     description: Track Network Flows
     template_path: flow.yml.hbs
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/icmp/manifest.yml:
```diff
@@ -6,6 +6,11 @@ streams:
     description: Capture ICMP Traffic
     template_path: icmp.yml.hbs
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/memcached/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/mongodb/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/mysql/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/nfs/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/pgsql/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/redis/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/sip/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented on packages/network_traffic/data_stream/thrift/manifest.yml:
```diff
@@ -3,6 +3,11 @@ type: logs
 streams:
   - input: packet
     vars:
+      - name: interface
```
This will be removed before merge, its to help with people testing, see PR description.
While we would have wanted this to be included, apparently packetbeat do not support different interfaces for different protocols, but it does work with pcaps.

P1llus commented:
Added new and better screenshots, Tests are most likely failing due to the test specific commit I added in, it will go back to green once its removed (once people have had the oppertunity to test)

P1llus commented:
Seems like some of the filters even references dataviews, will have to re-add all filters to remove any reference to `logs-*`

elasticmachine commented:
Package network_traffic - 1.16.0 containing this change is available at https://epr.elastic.co/search?package=network_traffic
